[RFR] Feature/page furniture

enginehouse opened the pull request:
Headers / footers / wrappers / logos / nav etc

Commit:
Added logo and svg

Commit:
Adding components for header

Commit:
Merge branch 'feature/typography-scales' into feature/page-furniture

# Conflicts:
#	package-lock.json

Commit:
Merge branch 'feature/179-contentful-setup' into feature/page-furniture

# Conflicts:
#	app/shared/routes.jsx
#	package-lock.json

Commit:
Button data and other attributes

Commit:
Reinstating the client side React

Commit:
Reinstating client side

Commit:
Linting

Commit:
Removing duplicate ‘this’ binding

Commit:
Experimenting with the routes

Commit:
Adding some off canvas menu stuff

Commit:
Additional components - toggle, grids, layout and longform

Koded committed:
All dependencies are development when using webpack bundling

Koded committed:
Updating configuration

Commit:
Drug page bits and pieces

Commit:
Merge branch 'develop' into feature/page-furniture

Commit:
Updating component with more / altered headings

Commit:
Merge branch 'develop' into feature/page-furniture

# Conflicts:
#	app/shared/routes.jsx
#	package-lock.json
#	package.json

enginehouse commented:
I think this is getting a little large and flabby. Going to put this in and then get more granular features branched in from now on in

Koded commented on app/shared/components/Button/component.jsx:
```diff
@@ -0,0 +1,61 @@
+    })
+
+    // list out data- and aria- attributes and stick em in
+    for (const prop in this.props) {
```
Out of interest - Is this a fix for certain attributes React doesn't support?

enginehouse replied:
It didn't like the aria- roles for some reason and kept rejecting it. The data attributes aren't really needed any more as we are not using Bootstrap javascript or jQuery so I should whip that out

enginehouse replied:
Oh also I would add that it is a plain old button - so this is the only way (I could think of) to get aria attributes only when needed

Koded replied:
ok makes sense

Commit:
Missed bad merge detritus

Commit:
Linting

Commit:
Adding svg and related components - updating toggle to new designs

Commit:
left margin on icons

Commit:
Name change for drug placeholder

Commit:
Image updated

enginehouse commented:
There is still some tinkering to be done here as the designs are not yet static - but it would be good to get some eyes on this and get it in soon

Commit:
Linting

This is a funny one - it is complaining of irregular whitespace and I can’t see it or disbale it in the linting @koded

enginehouse commented:
@koded could you have a look at this - it is complaining of irregular whitespace but I can neither see it nor disable it
![screen shot 2018-08-08 at 08 27 40](https://user-images.githubusercontent.com/5637994/43822849-429ff130-9ae5-11e8-9db0-97b31d6b11ac.png)

Commit:
Merge branch 'develop' into feature/page-furniture

Koded committed:
Merge branch 'feature/page-furniture' of github.com:PublicHealthEngland/talktofrank-alpha into feature/page-furniture

Koded committed:
Ignore eslint for jsx line

Koded committed:
Merge branch 'feature/page-furniture' of github.com:PublicHealthEngland/talktofrank-alpha into feature/page-furniture

Commit:
Increased the letter spacing

Koded committed:
Fixing bad merge

Commit:
Merge branch 'feature/page-furniture' of github.com:PublicHealthEngland/talktofrank-alpha into feature/page-furniture

Commit:
Kitting out the drug page with the correct header and new logo

Commit:
Content updates and some masthead alterations

Koded commented on app/server/index.jsx:
```diff
@@ -98,21 +96,21 @@ app.get('*', function (req, res) {
     let componentHead = ReactDOMServer.renderToStaticMarkup(<Head {...state.app.pageData} error={state.app.error} />)
-    let componentScripts = ReactDOMServer.renderToStaticMarkup(<Scripts />)
-    let renderedHtml = renderFullPageHtml(componentHtml, componentHead, componentScripts, JSON.stringify(state))
+    let renderedHtml = renderFullPageHtml(componentHtml, componentHead, JSON.stringify(state))
```
@enginehouse was this deliberate? It would be good to keep as much templating in React including things like scripts. Can you not do this in the `<Scripts />` component?

enginehouse replied:
It was, because the timestamp is being generated in `config-loader.js` and I figured I would keep it there as are the only js files we are using and are pretty constant - can easily shift them out though

Koded replied:
Ok - it was working - i.e. passing timestamp to the component, but let me double check it...

enginehouse replied:
I don't think there was any client side react in the version I had orginally, but like I say, I'll switch it back

Koded committed:
Reworking scripts

Commit:
Menu tinkering

Commit:
Merge branch 'feature/page-furniture' of github.com:PublicHealthEngland/talktofrank-alpha into feature/page-furniture

Commit:
Linting
